frontend/src/components/TransactionItem.tsx

```diff
@@ -90,21 +89,23 @@ function TransactionItem({ tx }: Props) {
               )}
             </div>
           </div>
-          <div className="overflow-hidden mr-3 flex flex-col items-start justify-center">
-            <div className="flex items-center gap-2 truncate">
-              <p className="text-lg md:text-xl font-semibold">
-                {type == "incoming" ? "Received" : "Sent"}
-              </p>
-              <p className="text-sm md:text-base truncate text-muted-foreground">
-                {dayjs(tx.settledAt).fromNow()}
+          <div className="overflow-hidden mr-3 max-w-full text-left">
+            <div>
+              <p className="flex items-center gap-2 truncate">
+                <span className="md:text-xl font-semibold">
+                  {type == "incoming" ? "Received" : "Sent"}
+                </span>
+                <span className="text-xs md:text-base truncate text-muted-foreground">
+                  {dayjs(tx.settledAt).fromNow()}
+                </span>
               </p>
             </div>
             <p className="text-sm md:text-base text-muted-foreground break-all flex">
-              {tx.description}
+              <p className="truncate">{tx.description}</p>
```

Comment:
try not to do nested `<p>` tags. I've found a way to fix the ellipsis not showing on the truncate